src/function/gds/wsp_destinations.cpp

```diff
@@ -0,0 +1,186 @@
+#include "binder/expression/node_expression.h"
+#include "function/gds/gds_function_collection.h"
+#include "function/gds/rec_joins.h"
+#include "processor/execution_context.h"
+#include "wsp_utils.h"
+
+using namespace kuzu::binder;
+using namespace kuzu::common;
+using namespace kuzu::storage;
+
+namespace kuzu {
+namespace function {
+
+class Costs {
+public:
+    Costs(const table_id_map_t<offset_t>& numNodesMap, MemoryManager* mm) { init(numNodesMap, mm); }
+
+    void pinTable(table_id_t tableID) { costs = costsMap.getData(tableID); }
+
+    double getCost(offset_t offset) { return costs[offset].load(std::memory_order_relaxed); }
+
+    void setCost(offset_t offset, double cost) {
+        costs[offset].store(cost, std::memory_order_relaxed);
+    }
+
+    bool update(offset_t boundOffset, offset_t nbrOffset, double val) {
```

Comment:
Can you add a comment here to clarify that this function updates the cost of `nbrOffset` to the one of `boundOffset` plus val, if the latter one is smaller. The function name is too brief to convey the actual computation happened.

Reply:
Done